### ISSUE TITLE
Modal created via ModalService is not hidden after calling .hide()

### ISSUE THREAD

**Issue by PascalHonegger** [valor-software/ngx-bootstrap#3711]
### Bug description or feature request:
Calling the hide() function on a modal will not hide it if it's not shown fully. This means that if I open a dialog to signal a usually long loading server call which finishes too fast (faster than the modal animation) the hide() will be called but has no effect. You can see this in my demo with three manual delays. At no or small delays the dialog just stays open, at 150ms it isn't even shown properly and at 250ms it works as expected. Another interesting behaviour is that opening many dialogs and closing them fast will leave the GUI unclickable (E.g. in my demo by clicking and holding ctrl+2).

### Plunker/StackBlitz that reproduces the issue:
StackBlitz: https://stackblitz.com/edit/ngx-bootstrap-ftrzc4?file=app%2Fapp.component.ts

### Versions of ngx-bootstrap, Angular, and Bootstrap:

ngx-bootstrap: 2.0.2

Angular: 4.4.6

Bootstrap: 3.3.7

**Comment by vizjerai** [comment]
This is still an issue with `ngx-bootstrap` v2.0.5 and v3.0.0

Setting the option `animated: false` helps but the race condition still exists.

**Comment by Roman-Simik** [comment]
Any progress, in which version will be this fixed ?

**Comment by dmitry-zhemchugov** [comment]
@Roman-Simik Fix is now in testing 
https://github.com/valor-software/ngx-bootstrap/pull/4550

**Comment by renaldas-kerpe-arria** [comment]
Which version was this fixed in? I see the same issue in 8.x.

Looks like this issue might have been re-introduced https://github.com/valor-software/ngx-bootstrap/issues/6195

**Comment by bakermeoe** [comment]
> This is still an issue with `ngx-bootstrap` v2.0.5 and v3.0.0
> 
> Setting the option `animated: false` helps but the race condition still exists.

how about using bsModalService .hide() function? will it better? so we still use animation and the modal will be hidden.

Edit: I just realized, ngx-bootstrap documentation doesn't inform this function. I saw the library code, there is .hide() function inside BsModalService class.
but the version of ngx-bootstrap is different with you, mine is v7.1.2

![image](https://user-images.githubusercontent.com/11452447/235829682-28d05761-a98a-4a96-b930-e8649f44991b.png)
